app/src/main/java/com/cornellappdev/android/eatery/ui/components/general/EateryCard.kt

```diff
@@ -374,54 +452,117 @@ fun EateryCardPrimaryHeader(eatery: Eatery, isCompact: Boolean) {
-                contentDescription = null,
-                tint = GrayFive,
-                modifier = Modifier.padding(end = 4.dp, top = 1.dp)
-            )
+            val openUntil = eatery.getOpenUntil()
             Text(
-                text = walkText,
-                fontSize = 14.sp,
-                fontWeight = FontWeight(500),
-                color = GrayFive,
-                style = EateryBlueTypography.subtitle2
+                modifier = Modifier.padding(top = 2.dp),
+                text =
+                if (openUntil == null) "Closed"
+                else if (eatery.isClosingSoon()) "Closing at $openUntil"
+                else ("Open until $openUntil"),
+                style = EateryBlueTypography.subtitle2,
+                color = if (openUntil == null) Red
+                else if (eatery.isClosingSoon()) Yellow
+                else Green
             )
 
-            val waitTimes = eatery.getWaitTimes()
-            if (!waitTimes.isNullOrEmpty()) {
-                DotSeparator()
-                Icon(
-                    Icons.Default.Schedule,
-                    contentDescription = null,
-                    tint = GrayFive,
-                    modifier = Modifier.padding(end = 4.dp, top = 1.dp)
-                )
-                Text(
-                    text = waitTimes,
-                    color = GrayFive,
-                    style = EateryBlueTypography.subtitle2
-                )
-            }
-
         }
     }
 }
 
+//TODO, integrate backend for the entree recommendation once backend is done for that
+@Composable
+fun EateryCardTertiaryHeader(eatery: Eatery, isCompact: Boolean, isGridView: Boolean) {
+    //TODO uncomment when backend finishes AI feature
+//    if (!isCompact && !isGridView) {
+//        Row(
+//            modifier = Modifier.padding(top = 2.dp),
+//            verticalAlignment = Alignment.CenterVertically
+//        ) {
+//            Text(
+//                text = "Recommended for You: ",
+//                color = GrayFive,
+//                style = EateryBlueTypography.subtitle2
+//            )
+//            Text(
+//                text = "Carved Roast Beef",
+//                color = EateryBlue,
+//                fontStyle = FontStyle.Italic,
+//                style = EateryBlueTypography.subtitle2
+//            )
+//        }
+//    }
+//    else{
+//        Column (modifier = Modifier.padding(top = 8.dp)){
+//            Text(
+//                text = "Recommended for You: ",
+//                color = GrayFive,
+//                style = EateryBlueTypography.subtitle2
+//            )
+//            Text(
+//                text = "Carved Roast Beef",
+//                color = EateryBlue,
+//                fontStyle = FontStyle.Italic,
+//                style = EateryBlueTypography.subtitle2
+//            )
+//        }
+//    }
+}
+
+//@Composable
+//fun EateryCardSecondaryHeader(eatery: Eatery, isCompact: Boolean) {
```

Comment:
If this won't be used in the future feel free to remove it, otherwise add the same TODO comment you've added to the other Composables here